### PR TITLE
fix: use sentinel on BulkWriter.delete()

### PR DESCRIPTION
**Pull request by thebrianchen**
Porting from [node](https://github.com/googleapis/nodejs-firestore/pull/1117).

Will add integration tests in separate PR.

### CODE REVIEW

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java [hunk 1]
```diff
@@ -697,12 +697,21 @@ private T performDelete(
...
+              // TODO(b/158502664): Use actual delete timestamp.
+              boolean isSuccessfulDelete =
+                  !writeResult.hasUpdateTime()
+                      && Status.fromCodeValue(status.getCode()) == Status.OK;
+              Timestamp DELETE_TIMESTAMP_SENTINEL = Timestamp.ofTimeSecondsAndNanos(0, 0);
+              if (isSuccessfulDelete) {
+                updateTime = DELETE_TIMESTAMP_SENTINEL;
+              } else if (writeResult.hasUpdateTime()) {
+                updateTime = Timestamp.fromProto(writeResult.getUpdateTime());
+              }
```
Comment (schmidt-sebastian) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java:713]: This entire block is just:

```
Timestamp updateTime = status.isOk() ? Timestamp.fromProto(writeResult.getUpdateTime()) : null;
```

getUpdateTime() defaults to an empty Timestamp in the Java API.

Comment (thebrianchen) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java]: done. It's pretty much just comment changes at this point

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java [hunk 2]
```diff
@@ -697,12 +697,11 @@ private T performDelete(
...
-                  new BatchWriteResult(
-                      writeResult.hasUpdateTime()
-                          ? Timestamp.fromProto(writeResult.getUpdateTime())
-                          : null,
-                      Status.fromCodeValue(status.getCode())));
+              Timestamp updateTime =
+                  Status.fromCodeValue(status.getCode()) == Status.OK
+                      ? Timestamp.fromProto(writeResult.getUpdateTime())
+                      : null;
+              result.add(new BatchWriteResult(updateTime, Status.fromCodeValue(status.getCode())));
```
Comment (BenWhitehead) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java:704]: nit: Duplicate `Status.fromCodeValue...` from line 701

Comment (thebrianchen) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java:704]: thanks, done.

**Pull request merged**